src/app/components/board/board.component.ts

```diff
@@ -43,11 +43,13 @@ export class BoardComponent implements OnInit {
     this.isThereAWinner = false;
   }
 
-  onClick(y: number) {
-    const x = this.getXPosition(y);
-    if (x !== undefined) {
-      this.paintPiece(x, y);
-      this.lastTurnCoordenates = {x, y};
+  onClick(x: number, y: number) {
+    const xFall = this.getXPosition(y);
+    if (xFall !== undefined) {
+      const startPosition: Coordenate = {x: x + 1, y};
+      const endPosition: Coordenate = {x: xFall, y};
+      this.animateFall(startPosition, endPosition);
+      this.lastTurnCoordenates = {x: xFall, y};
```

Comment:
The endPosition and lastTurnCoordinates have the same value. You can just assign endPosition to lastTurnCoordinates